streamz/core.py

```diff
@@ -181,6 +181,24 @@ def _inform_asynchronous(self, asynchronous):
                 if downstream:
                     downstream._inform_asynchronous(asynchronous)
 
+    def _add_upstream(self, upstream):
+        if self.upstreams == [None]:
+            self.upstreams = [upstream]
```

Comment:
Any reason to make a new list, instead of `self.upstreams[0] = upstream`?
I suppose you were just copying how the code looked previously.